### PR TITLE
all needed controllers and filter implemented

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (allowedUrls.contains(url)) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java]: can our service return null?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
nice job, almost done

File: src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java]: additional null check removed

File: src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector = Injector.getInstance("core.basesyntax");
+    private final Set<String> allowedUrls = new HashSet<>();
+    private final DriverService driverService = (DriverService) injector
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java]: we don't use it as well as injector remove both of these fields

**Review by RomanDubovskyi — APPROVED** [review]
remove redundant fields in the filter before a merge

File: src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java]: done

**Pull request merged**